### DISCUSSION TITLE
kwh meter not working

### DISCUSSION THREAD

**Discussion by AppleFlapp** [huizebruin/s0tool#128] [General]
Hello,

Not sure if this is the right place for this but I give it a shot.

Yesterday I started using the pulse meter function of the S0 tool. But I have a few issues.
kWh sensor status is going nuts switching on and off and on and off and...

Another issue is that the entity Actuele Energie is not stable. Even when there is no load the value fluctuates. 
Pull up resistor also installed. This already did improve the flutcation of the Actuele Energie but it is still not ok.

And nothing changes when there is a load. Actuele Energie keeps doing what was described above.
Actuele meterstand stays on what it is set to.

When I disconnect the + line to the kWh meter pulse output the Actuele energie is counting up as well as the 
Actuele meterstad but still without load. 

I am using the Eastron SDM120D MID.

Programming the ESP was done via the website. 

I use the kWh meter to read out the use of an outlet socket.

Anyone who can help? I'm not an expert in this yet.

I see a few possible options at the moment.
1. Something in the code is wrong.
2. ESP board pin D5 is end of life.
3. The Eastron pulse output is not working.

**Comment by AppleFlapp** [comment]
Anyone who can point me in the right direction?

**Comment by AppleFlapp** [comment]
No one here? :(

**Reply by huizebruin** [reply]
i dont get any message when somting is post here.
i will look for your question today and will com back with a anser.

**Reply by AppleFlapp** [reply]
Thanks. Any help is appreciated.
It can also be done in Dutch but then the audience is smaller.

**Reply by huizebruin** [reply]
For Dutch help, look for my discord server. On my website. So we can chat and help to figure it out. There are 210 people on it.

**Reply by AppleFlapp** [reply]
Were you able to find something in the code or do you have any other ideas?

**Reply by huizebruin** [reply]
Did you order the S0tool or did you build it yourself?

On My S0tool I placed a ceramic condensator and a normal condensator.
To filter some of the spikes.

Some meters are giving a spike when they are working. Can't find why and how to filter it in the code.

On my meter I don't know it. And it's connected to the S0tool too.

**Reply by AppleFlapp** [reply]
I did build it myself. It's not that complicated. 

I already thought to see those on the picture of your device but you are not showing those in any wiring diagram on your site.
Can you give more information about this so that I can updated my device?

Btw, I never said the code is the problem. It was just one of my three options.
And before I start buying replacement hardware to test, maybe even without success, I was asking for help here.

**Reply by huizebruin** [reply]
Can you send me an email from my website. I will help you with a Dutch text.

**Reply by AppleFlapp** [reply]
Will do that as soon as possible. I'm currently not in the position to do so.

**Reply by huizebruin** [reply]
How to update the S0tool.

https://github.com/huizebruin/s0tool/issues/135#issuecomment-1826302056

**Reply by AppleFlapp** [reply]
Thanks for the information. 

I wanted to contact you via mail as you suggested a long time ago but I was very busy since and also some serious health issues in the family. Luckily that is now improving. But I need to dig in it again to see what the problem was and if I can fix it without your help.
If not, I will still send you a mail. Might take a while though...

**Reply by huizebruin** [reply]
Or u can come to my Dutch discord group, about HA and the s0tool and my website and more. 
There are 240+ people on it talking about all above. 
See my website for the discord link.

**Comment by HobbitBluelabel** [comment]
😅



Op vr 8 sep 2023 om 10:30 schreef AppleFlapp ***@***.***>:
